lib/cn/csched_sp3.c

```diff
@@ -320,6 +317,25 @@ struct sp3 {
 #define debug_qos(_sp)         (csched_rp_dbg_qos((_sp)->rp))
 #define debug_rbtree(_sp)      (csched_rp_dbg_rbtree((_sp)->rp))
 
+static inline bool
+qfull(struct sp3 *sp, uint qnum)
+{
+    struct sp3_qinfo *qi = sp->qinfo + qnum;
+
+    return qi->qjobs >= qi->qjobs_max;
+}
+
+static inline uint
+qthreads(struct sp3 *sp, uint qnum)
+{
+    uint64_t rparam = sp->rp->csched_qthreads;
+    uint n;
+
+    n = (rparam >> (qnum * 8)) & 0xff;
+
+    return clamp_t(uint, n, 1, 16);
+}
```

Comment:
I put some limits on the queue size to keep users from shooting themselves in the foot.  And we need at least one thread per queue to prevent a scheduling stall..

Reply:
Do we use this feature (changing thread counts at runtime)?  Might just lose it.  I'd rather redo how we select and run compaction jobs than add logic to increase/decrease number of compaction threads.